packages/monorepo-releaser/index.js

```diff
@@ -0,0 +1,147 @@
+ * @module monorepo-releaes
+ */
+
+/**
+ * Semantic Release branch config object
+ * {@link https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#branches-properties|sementic-release branches object}
+ * @typedef {object} semrelBranches
+ * @property {string} name - branch name
+ * @property {string} [channel] -
+ *    {@link https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#channel|distribution channel}
+ *    property used to modify the NPM distribution tag name
+ * @property {string} [range] - regex-string used for number-based releases
+ * @property {string} [prerelease] - used in pre-release version naming
+ * @property {boolean} [main] - undocumented config variable which tells semantic-release/github that a branch is `main` so releases
+ *    on that branch are not labeled `prerelease`
+ */
+
+/**
+ * Semantic Release options object. See
+ * {@link https://semantic-release.gitbook.io/semantic-release/usage/configuration#branches|semantic-release options object}
+ * note: this typedef does not contain all avilable options from semantic-release, just PAL-relevant properties.
+ * See semantic-relase docs for more config options.
+ * @typedef {object} semrelOptions
+ * @property {object} repositoryUrl - from package.json
+ * @property {object} githubUrl - root URL for Github - different on enterprise GitHub
+ * @property {object} githubApiPathPrefix - API path for Github - different on enterprise GitHub
+ * @property {object} [releaseNotes] - strings and data used when writing release notes
+ * @property {semrelBranches[]} [branches] - release branches/channels config
+ *  {@link https://semantic-release.gitbook.io/semantic-release/usage/configuration#branches|config.branches}
+ * @property {object} [plugins] - semantic-release plugins to use.
+ *  {@link https://semantic-release.gitbook.io/semantic-release/usage/configuration#plugins|config.plugins}
+ *  default: [
+ *      '@semantic-release/commit-analyzer',
+ *      '@semantic-release/release-notes-generator',
+ *      '@semantic-release/npm',
+ *      '@semantic-release/github'
+ *    ],
+ */
+
+/**
+ * This object contains all configurations which are unique to the test-automation repository
+ * @type {object}
+ * @property {semrelBranches} branches - test-automation branches configuration, different from semantic-release
+ *    default branches config by exchanging `master` with `develop`
+ * @property {string[]} packages - repo packages that should be released on merge, in order of dependency use. This
+ *    array is to configure the order in which the packages should be released to npm. This array should include
+ *    all packages which are released on merge. These are ordered to ensure a dependency is released
+ *    before any module which will need to replace the wildcard with the latest version number.
+ *    The wildcards are replaced via semantic-release-wildcards
+ */
+const testAutomation = {
+  branches: [
```

Comment:
For my own curiosity, are we actually planning on having all of these branches?  Right now it looks like we only have `develop`

Reply:
not sure really. semantic-release fails if a configured `release` branch is not already on the gh-repo. So instead of adding `master` I changed the default config, switching in `develop` to replace `master`
There is no plan for branch naming of this repo so I opted to stay as close as possible to semantic-release defaults to make it as familiar as possible to use.

Reply:
>  if a configured release branch is not already on the gh-repo

Wouldn't this mean that it would fail because none of these other branches are created?  I guess I was expected something as simple as `branches: ['develop']` since all we have is that branch

Reply:
no, it only requires only one of the release branches to be set up. The error is 
`A minimum of 1 and a maximum of 3 release branches are required in the branches configuration.`

it is also mentioned in the [branches config docs](https://semantic-release.gitbook.io/semantic-release/usage/configuration#branches)

```
Note: If your repository does not have a release branch, then semantic-release 
will fail with an ERELEASEBRANCHES error message. If you are using the default
configuration, you can fix this error by pushing a master branch.
```